### PR TITLE
Refactored callbacks in async array iterations

### PR DESCRIPTION
**Pull request by tshemsedinov**

### CODE REVIEW

File: lib/array.js [hunk 1]
```diff
@@ -3,66 +3,81 @@
...
     let count = 0;
     items.forEach((item, index) => {
-      callback(item, (err, value) => {
+      fn(item, (err, value) => {
+        if (errored) return;
         if (err) {
-          hadError = true;
-          return done(err);
+          errored = true;
+          if (done) done(err);
```
Comment (aqrln) [inline lib/array.js]: Nit: I'd prefer this function to be called `onDone` or `callback`. The current name makes conditions like this one misleading.

File: lib/chain.js [hunk 2]
```diff
@@ -31,8 +31,8 @@ module.exports = (api) => {
     }
 
     filter(fn) {
-      this._update((array, resolve) => {
-        api.metasync.filter(array, fn, result => resolve(result));
+      this._update((array, resolve, reject) => {
```
Comment (aqrln) [inline lib/chain.js]: Note that all these methods can now be greatly simplified. I supposed them all to be one-liners, but inconsistent callback style impeded it.

Comment (tshemsedinov) [inline lib/chain.js]: Yes, will refactor ASAP

Comment (aqrln) [inline lib/chain.js]: You may do that in this PR or another one, I don't feel strongly for either option. Keep in mind though that it'll need a new round of review if you decide to update this PR.

Comment (tshemsedinov) [inline lib/chain.js]: I started #50 having in mind exactly this :smile: but want to merge #59 before

**Comment by tshemsedinov** [comment]
Landed in a71af4f

**Pull request closed without merging**